starter.go

```diff
@@ -48,15 +48,20 @@ func (s *starter) manipulate(c *chunk) outcome {
 	starts := epochMillisecond(c.start)
 	ends := epochMillisecond(c.end.Add(-time.Millisecond)) // CWL uses inclusive time ranges, we use exclusive ranges.
 
+	// Convert Limit from int64 to int32 for AWS SDK Go v2
+	limit := int32(c.stream.Limit)
```

Comment:
Rather than doing this, it feels safer to change the data type in the field `QuerySpec.Limit` from `int64` to `int32`.

A straight cast here is risky, because it's changing the data type without accounting for possible overflow.

https://go.dev/play/p/yEfg_AY270T

Yes, technically, we check it against `MaxLimit` but the best way to do things is to maintain the same type all the way through the API to reduce the possibility of error.

FWIW, `int64` was chosen previously because that was the value accepted by AWS SDK for Go V1.